### DISCUSSION TITLE
Android Support for Maia SDR

### DISCUSSION THREAD

**Discussion by jacobagilbert** [maia-sdr/maia-sdr#7] [General]
@daniestevez Thanks for publishing this tool, its very useful what you have done!

I just wanted to convey my experiences with getting this working on android in the hope that it is of value to others, and that maybe someone knows more about the issues I have seen and possibly workarounds. I tried this on three android phones with different quirks, a Pixel 3 running Android 12 and a Pixel 6a and Pixel 7 both running Android 13.

1. Android 13 seems to treat USB OTG devices differently than Android 12. In the case of the Pixel 3 I was able to use a simple USB-C OTG to micro-B cable and have the USB ethernet network interface appear (`eth0` in Android) and get the appropriate IP address from the Pluto. For both the Pixel 6a and 7 on Android 13, I needed to **use a USB Hub inline with the Pluto** in order for this interface to appear, and interestingly a port breakout device like this one https://www.amazon.com/dp/B08TWKNV13 did not work directly, I needed to add a discrete USB hub. No configuration settings seemed to fix this; for example, I would get a warning that I "Couldn't Switch" if I tried to control the USB connection by the phone:
<img width="214" alt="image" src="https://user-images.githubusercontent.com/7121282/219683206-781b31db-c0a0-49a1-ab43-14a88857bf67.png">

1. After ensuring `eth0` was present and valid in step 1, in all cases for me it was necessary to either enable Android's `Ethernet Tethering` feature, or to turn off all other network interfaces (WiFi, Cellular Data) in order to be able to communicate (ping, http traffic, etc) with the Pluto. I suspect this has something to do with the way Android routing is setup for this type of interface but I don't know of any workarounds. Because of this its important to pay attention to the `fw_setenv ipaddrmulti 1` option for the Pluto (see [installation guide](https://maia-sdr.org/installation/)); this is somewhat annoying as you must first determine which IP address will be used by Android for the connection. I was unable to get `pluto.local` to resolve which would alleviate that annoyance.

I wish I knew more about Android to trace some of this down to the root cause, but hopefully these notes are useful for others.

**Comment by daniestevez** [comment]
Hi @jacobagilbert, thanks for the tests and the information. I think this is good content for the wiki. I've added a [wiki page](https://github.com/maia-sdr/maia-sdr/wiki/Android-support), where I've already written the issues and tricks I found with my phone. Feel free to add the information about the devices you've tested (all Github users should have edit access to the wiki, let me know if this doesn't work).

I will link this wiki page from the installation instructions so that folks can find it easily.

There is also the option of setting up the phone as a device and the Pluto as host. I haven't explored this possibility, mainly because I don't have the OTG cable to do that, but this approach could be a solution for cases in which the other set up fails.

I don't know much about Android either, so I was glad to get this working, even if in a slightly inconvenient way.
